### PR TITLE
fix(braindump): complete plain prose lines, not just checkbox lines

### PR DESCRIPTION
**Pull request by ryota-murakami**
## Summary

The BrainDump complete command (Cmd/Ctrl+Enter) only worked on lines that were already formatted as `- [ ]` markdown checkboxes. An ordinary prose line could never be logged as Completed — pressing Cmd/Ctrl+Enter on it did nothing. This broadens the command so any non-empty line can be completed with the natural keyboard gesture, no markdown pre-typing required.

## The bug (root cause)

`handleKeyDown` parsed the caret line and bailed on anything that wasn't a checkbox:

```ts
const parsed = parseCheckboxLine(line, lineIndex)
if (!parsed) return   // ← plain prose lines died here; create mutation fired 0 times
```

Confirmed by reproduction: a plain line + Cmd/Ctrl+Enter fired the Completed-create mutation **zero** times, while a `- [ ]` line fired it once. So the row was never created — a write-path scope gap, not a display issue.

## The fix

Replace the bail with a plain-line completion branch. A new pure helper `markPlainLineCompleted` wraps a non-checkbox line as `- [x] <text>` and returns the rewritten note + the title to persist, mirroring the checkbox path's uncapped-title behavior (promote caps it for the DB). It guards the no-op cases:

- already well-formed checkbox lines → `null` (the existing toggle path owns them)
- empty checkbox skeletons (`- [ ]`, `- [x]`, `- []`) → `null` (nothing to record)
- blank lines and out-of-range indices → `null`

Placeholder and JSDoc now describe the broadened command instead of the old checkbox-only behavior.

## Tests

- `braindumpUtils.test.ts`: 7 specs for `markPlainLineCompleted` (wrap, multi-line caret targeting, trim, and all four null guards).
- `BrainDumpEditor.test.tsx`: 3 complete-command specs — plain prose line → Completed row, existing checkbox still toggles (no regression), blank line is a no-op. Mock refactored to a shared hoisted spy so the create call can be asserted.

## Verification

- `pnpm test` → **585 passed | 29 skipped** (fresh)
- `pnpm build` → compiled successfully (fresh)
- `pnpm lint` + `pnpm typecheck` → green

## Design alignment

Serves the DESIGN.md north star (些細でも経験値 / self-affirmation, low friction): completing any non-empty line with the natural Cmd/Ctrl+Enter gesture — without forcing users to pre-type `- [ ]` markdown just to log a win.

## Summary by CodeRabbit

* **New Features**
  * Users can now complete the current plain text line with Cmd/Ctrl+Enter; it’s converted into a completed checkbox item while preserving the original text as the title.
  * Editor instructions/placeholder now clarify completing the current line via the keyboard shortcut.

* **Bug Fixes**
  * Improved completion failure handling: if creation fails, the editor restores the original line content without affecting unrelated lines.

* **Tests**
  * Added/extended coverage for plain-line completion, conversions, and rollback edge cases.

### CODE REVIEW

File: src/components/braindump/BrainDumpEditor.tsx [hunk 1]
```diff
@@ -571,7 +574,17 @@ export const BrainDumpEditor = memo(function BrainDumpEditor({
...
+    if (!parsed) {
+      // Not a checkbox line: let the complete command finish an ordinary prose
+      // line by wrapping it as `- [x] …` and promoting it, so users don't have
+      // to pre-type `- [ ]` markdown just to log a win. Blank lines and empty
+      // checkbox skeletons return null and fall through to a no-op.
+      const promoted = markPlainLineCompleted(text, lineIndex)
+      if (!promoted) return
+      setNoteText(promoted.text)
+      void promoteLineToCompleted(lineIndex, promoted.title)
+      return
```
[the remark this answers is not in the document]
Comment (ryota-murakami) [reply src/components/braindump/BrainDumpEditor.tsx:621]: Fixed in f4f13f3 — but I diverged from the literal `setNoteText(rollbackText)` suggestion to avoid a data-loss regression. ✅

**Why not the literal fix:** `rollbackText` is the snapshot captured *before* the async create. This error path deliberately reads `noteTextRef.current` (latest) and re-resolves the line by title precisely because the user can keep typing while the create is in flight (see the existing drift comment). Replaying the stale snapshot would discard those concurrent edits.

**What I did instead:** a surgical `replaceLineAtIndex(text, line, prose)` helper + a `rollbackPlainText?` param on `promoteLineToCompleted`. The prose-restore runs **only when the title search positively finds the line**; on a miss it falls back to the existing safe uncheck (a no-op on non-checkbox lines), so a drifted stale index can never blind-overwrite an unrelated line the user just edited. Covered by a RED-proven miss-branch regression test.

**Deferred:** the manual-undo (toast "Undo") plain-line path still reverts to `- [ ]`; that belongs with the upcoming keep-note-vs-clear-on-complete work and is out of scope here.
